org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/FormatterHandlerTest.java

```diff
@@ -224,4 +229,34 @@ public void testDocumentFormattingWithCustomOption() throws Exception {
 		String newText = TextEditUtil.apply(unit, edits);
 		assertEquals(expectedText, newText);
 	}
+
+	@Test
+	public void testGoogleFormatter() throws Exception {
+		try {
+			String text =
+			//@formatter:off
+					"package org.sample;\n\n" +
+					"public class Baz {\n" +
+					"  String name;\n" +
+					"}\n";
+				//@formatter:on"
+			ICompilationUnit unit = getWorkingCopy("src/org/sample/Baz.java", text);
+			String uri = JDTUtils.toURI(unit);
+			TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+			FormattingOptions options = new FormattingOptions(2, true);// ident == 2 spaces
+			DocumentFormattingParams params = new DocumentFormattingParams(textDocument, options);
+			Bundle bundle = Platform.getBundle(JavaLanguageServerTestPlugin.PLUGIN_ID);
+			URL googleFormatter = bundle.getEntry("/testresources/eclipse-java-google-style.xml");
+			URL url = FileLocator.resolve(googleFormatter);
+			preferenceManager.getPreferences().setFormatterUrl(url.toExternalForm());
```

Comment:
can you add a test for non-uri values, that should actually resolve to file uri?